doc/api/errors.md

```diff
@@ -794,6 +794,12 @@ An invalid symlink type was passed to the [`fs.symlink()`][] or
 
 An attempt was made to add more headers after the headers had already been sent.
 
+<a id="ERR_HTTP_INCOMING_SOCKET_ENCODING"></a>
+### ERR_HTTP_INCOMING_SOCKET_ENCODING
+
+An attempt was made to manipulate the encoding of an incoming request packet.
+This is forbidden by [RFC7230 Section 3][rfc7230-3]
```

Comment:
I would word this a bit differently... Perhaps... 

```
An attempt was made to set the text encoding of a `Socket` bound to
an HTTP request/response. This is not permitted by HTTP.
```

Reply:
```
An attempt to set text encoding of a HTTP request or response `Socket`
is not permitted. [RFC7230 Section 3][rfc7230-3]
```
A bit less passive?
I think keeping a reference to _why_ it's not permitted is good. It could help developers get an immediate and relative answer without poking around for the _why_. But it is the first documented error to include the RFC reference which would likely lead to more RFC specific errors having RFC article/sections cited, or, I will remove the RFC citation and keep the status quo. Thoughts?